### PR TITLE
.Net: Nl2Sql Console @ Experimental data/nl2sql

### PR DESCRIPTION
**Pull request by crickman**
## What
A stand-alone natural language query console as an emulation/homage to query consoles such as sqlcmd.exe (SQL Server), mysqla.exe (MySQL), sqlplus.exe (Oracle), psql.exe (PostgreSQL).

## Goals:
- Enable experimentation/understanding of the NL2SQL space (how it works, where it works well, limitations, best-practices, etc...)
- Development "sandbox" - allow developers to target their own databases for initial POC.

## Non-Goals:
- Introduce re-usable "Connector"
- Introduce re-usable data-frame "Adapters"
- Support "Planner" integration/usage

## Details:
- Only target SQL Server (target MySQL for connector build out, not initial sample).
- Target AdventureWorksLT sample database ("out of the box").
- Developers can configure to target a different database/schema. 
- Support local and cloud hosted databases

### CODE REVIEW

File: samples/data/nl2sql/Readme.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+1. Demonstrate the natural ability of LLM to reason over an objective and generate a SQL query.
+1. Allow exploration of any *(SQL Server)* database (on-premises or cloud hosted).
+
+## Sample Info
+
+The default configuration targets two sample schemas, but it may be configured to target your own database as well.
+
+This sample is organized as follows:
+
+- `nlsql.config` - Contains [data-schemas](./nl2sql.config/schemas/Readme.md) and [semantic-prompts](./nl2sql.config/nl2sql/Readme.md).
```
Comment (dluc) [inline samples/data/nl2sql/Readme.md:21]: nit: rename to `nl2sql.*` maybe?

Comment (crickman) [reply samples/data/nl2sql/Readme.md:21]: Just a typo, naming is `nl2sql.*`....fixed.

**Comment by shawncal** [comment]
@crickman This project is big and independent enough, it probably belongs in its own repo. @alliscode or I can help set that up for you.

**Review by shawncal — CHANGES_REQUESTED** [review]
Let's move this to a feature repo

**Comment by crickman** [comment]
Moved to https://github.com/microsoft/semantic-sql-sandbox

**Pull request closed without merging**